app/controllers/users_controller.rb

```diff
@@ -11,16 +11,17 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t "common.welcome"
-      reset_session
-      log_in @user
-      redirect_to @user
+      UserMailer.account_activation(@user).deliver_now
+      flash[:info] = t "users.mailer.account_activation.check_email"
+      redirect_to root_url
     else
       render :new, status: :unprocessable_entity
     end
   end
 
-  def show; end
+  def show
+    redirect_to root_url and return unless @user.activated?
+  end
```

Comment:
```suggestion
  def show; end
```
Code bên trong e đưa ra method check_user_activated? và call trong before_action nha